### PR TITLE
fix: change sql to improve the performance of catalog api

### PR DESCRIPTION
**Pull request by chlins**
Close：https://github.com/goharbor/harbor/issues/14819

Signed-off-by: chlins <chlins.zhang@gmail.com>

### CODE REVIEW

File: src/pkg/repository/dao/dao.go [hunk 1]
```diff
@@ -160,7 +159,7 @@ func (d *dao) NonEmptyRepos(ctx context.Context) ([]*models.RepoRecord, error) {
 		return nil, err
 	}
 
-	sql := fmt.Sprintf(`select distinct r.* from repository as r LEFT JOIN tag as t on r.repository_id = t.repository_id where t.repository_id is not null;`)
+	sql := `select * from repository where repository_id in (select distinct repository_id from tag);`
```
Comment (heww) [inline src/pkg/repository/dao/dao.go:162]: `;` in this sql can be removed.

Comment (chlins) [reply src/pkg/repository/dao/dao.go:162]: ok

**Comment by chlins** [comment]
@wy65701436 Can you help look at it?  pls.

**Comment by chlins** [comment]
Before
`avg=7.32s    min=133.19µs med=6.85s    max=14.26s   p(90)=12.19s   p(95)=12.91s`

After
`avg=3.55s    min=153.61µs med=3.12s    max=7.18s    p(90)=5.77s    p(95)=6.48s`

**Review by wy65701436 — APPROVED** [review]
lgtm

**Review by heww — APPROVED** [review]

**Pull request merged**